lib/pry/code.rb

```diff
@@ -339,7 +339,7 @@ def method_missing(method_name, *args, &block)
         super
       end
     end
-    undef =~
+    undef =~ if instance_methods.include?(:=~)
```

Comment:
Will this work? Or maybe your approach has benefits?

```suggestion
    undef =~ if method_defined?(:=~)
```

Reply:
Actually yes, that works.  We changed method_defined? API several years ago.  I thought there were compatibility issues.  However that change does not affect this particular situation.  Was my overthinking.